src/helpers/SparseCalldataSegmentLib.sol

```diff
@@ -0,0 +1,41 @@
+// SPDX-License-Identifier: GPL-3.0
+pragma solidity ^0.8.25;
+
+library SparseCalldataSegmentLib {
```

Comment:
Considering the importance of this lib and the potential for errors with recursion, maybe we should add some fuzz tests.

Reply:
Added a fuzz test for the library here: https://github.com/erc6900/reference-implementation/blob/94a3764235338716d2ad2b38cfcd3501c98e8cc0/test/libraries/SparseCalldataSegmentLib.t.sol